marshmallow/fields.py

```diff
@@ -1259,14 +1261,14 @@ def _serialize(self, value, attr, obj, **kwargs):
             }
 
         # Serialize values
-        # Note: the dict type (dict, OrderedDict,...) of the value is lost
+        result = self.map_type()
         if self.value_container is None:
-            result = {keys[k]: v for k, v in iteritems(value) if k in keys}
```

Comment:
You could keep these comprehensions by passing the pairs to the `map_type`.

```py
result = self.map_type((keys[k], v) for k, v in iteritems(value) if k in keys)

Reply:
Tried similar approach in #1093 but as @sloria pointed out, i think there will be additional overhead of creating tuple for each pair.

Reply:
Yeah, creating tuples incurs a ~2x performance penalty.

<img width="500" alt="image" src="https://user-images.githubusercontent.com/2379650/51002648-e4c1c200-156e-11e9-907e-99586830e54c.png">